src/tsc/tsc.ts

```diff
@@ -174,15 +165,32 @@ namespace ts {
         }
     }
 
+    export function executeCommandLine(args: string[]): void {
+        if (args.length > 0 && args[0].charCodeAt(0) === CharacterCodes.minus) {
+            const firstOption = args[0].slice(args[0].charCodeAt(1) === CharacterCodes.minus ? 2 : 1).toLowerCase();
+            if (firstOption === "build" || firstOption === "b") {
+                return performBuild(args.slice(1));
+            }
+        }
+
+        const commandLine = parseCommandLine(args);
+
+        if (commandLine.options.generateCpuProfile && sys.enableCPUProfiler) {

```

Comment:
Error host doesn't support option if `!sys.enableCPUProfiler` ?

Reply:
Should it? It's essentially just a diagnostics flag.